src/wallet/wallet.cpp

```diff
@@ -2198,12 +2198,12 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
                     }
                     else
                     {
-                        if (nChangePosInOut == -1)
+                        if (nChangePosInOut < 0)
```

Comment:
Why change this?

Reply:
The cast is for the unsigned comparison(s), not changing the type.

Reply:
I mean static_cast<size_t>(-2) will result in insert container past end. line: 2206

Reply:
Ah, forgot it. Check is mistaken.